### ISSUE TITLE
Reset without having to power cycle?

### ISSUE THREAD

**Issue by tcushing** [Normalised/ProkDrumsEditor#20]
Seems as I'm using the editor sometimes the module will freeze up or I lose connection - and need to reset it.  Is there a more graceful way to do this rather than brute force power cycling?  I lose my changes doing that as well.

I'm considering making a separate switched power bus to toggle the +/- 12v rails to be able to individually power cycle a module if need be.  Not feeling too comfortable with this approach however with the digital Teensy boards.

Thanks in advance.

**Comment by Normalised** [comment]
The only 'remote' way I know of triggering a reboot is via the TY tools.

https://github.com/tyzjames/ty#tyc_reset

If you do use the TY tools then you have to take full responsibility if something happens to the modules.

I'd be interested to know if there are any particular circumstances that cause a freeze or lost connection.

**Comment by tcushing** [comment]
Thanks for the reply.  I was finding with the Mac Prok Editor the target module would sometimes freeze after sending a bank - I could no longer trigger it.  I'll see if I can narrow down the circumstances and replicate the problem.

Ultimately I was trying to work out the procedures for editing, saving, and loading patches - the manual isn't very straightforward (I realise it's to be updated).   Can I suggest a YouTube video explaining the ProkEditor functionality as a tutorial?  It would go a long way with helping the user base.

Cheers

**Comment by Normalised** [comment]
Yeah the bank functionality in the editor is pretty flaky. The situation will improve when i get the next set of updates done. Which is supposed to be now but sometimes life gets in the way :)

**Comment by tcushing** [comment]
Cheers for that - appreciate the support!

One very basic question I have - are the factory patches stored in ROM on the module?   Is there a danger of overwriting the factory patches - or does it always revert back to factory default when rebooted?  I'm concerned I'll damage the factory patches as I experiment with patch editing, saving, and loading in the editor.

Thanks again.

**Comment by Normalised** [comment]
factory patches are built into the firmware. no way of overwriting them.

the only way to store custom patches on the module and have them come back after a reboot is to use and SD card and hold the button when rebooting to load from SD.

But this SD card handling will be improved in the next update.

**Comment by tcushing** [comment]
Ok thanks for the info
